docs/sanity-migration/09-neon-integration.md

```diff
@@ -0,0 +1,487 @@
+# Chapter 9: Neon Integration
+
+[← Back to Index](./README.md)
+
+---
+
+## Overview
+
+Contest features (Top 11, Year End Poll, Modern Rock Madness) use a **hybrid architecture** that splits data between Sanity CMS and Neon PostgreSQL for optimal cost, performance, and functionality.
+
+### Why a Hybrid Approach?
+
+**Sanity** excels at:
+- Staff-managed content with editorial workflows
+- Visual content management interface
+- Reference relationships between content types
+- Versioning and publishing workflows
+
+**Neon (PostgreSQL)** excels at:
+- High-volume transactional data (thousands of votes)
+- Complex aggregation queries
+- Unique constraints for duplicate prevention
+- Storing personally identifiable information (PII)
+
+### Data Separation
+
+| Data Type | Storage | Rationale |
+|-----------|---------|-----------|
+| Contest configuration (dates, prizes, options) | Sanity | Staff-managed content, editorial workflow |
+| Song/album/artist options | Sanity | Reference existing content documents |
+| Voting windows and rules | Sanity | CMS-managed configuration |
+| Individual votes | Neon | High volume, API cost efficiency |
+| Write-in submissions | Neon | User-generated text data |
+| Contest entries (contact info) | Neon | PII, prize drawing queries |
+
+---
+
+## Architecture
+
+### Data Flow: Voting
+
+```
+┌─────────────┐
+│   Browser   │
+└──────┬──────┘
+       │ 1. User votes
+       ▼
+┌─────────────────┐
+│  API Endpoint   │ ← 2. Fetch contest config from Sanity
+│  (Node.js/PHP)  │ → 3. Validate vote
+└──────┬──────────┘
+       │ 4. Store vote
+       ▼
+┌─────────────┐
+│    Neon     │
+│ (PostgreSQL)│
+└─────────────┘
+```
+
+### Data Flow: Results
+
+```
+┌─────────────┐
+│    Neon     │
+│ (PostgreSQL)│ ← 1. Query vote counts
+└──────┬──────┘
+       │ 2. Aggregate results
+       ▼
+┌─────────────────┐
+│   Staff Tool    │ → 3. Publish results to Sanity
+└─────────────────┘
+       │
+       ▼
+┌─────────────┐
+│   Sanity    │ ← 4. Create top11Result document
+└─────────────┘
+       │
+       ▼
+┌─────────────┐
+│   Website   │ ← 5. Display published results
+└─────────────┘
+```
+
+---
+
+## Neon Schema Reference
+
+### Tables
+
+#### `users`
+Registered users and voters.
+
+| Column | Type | Description |
+|--------|------|-------------|
+| `id` | UUID | Primary key |
+| `email` | VARCHAR(255) | Unique email address |
+| `first_name` | VARCHAR(64) | First name |
+| `last_name` | VARCHAR(64) | Last name |
+| `phone` | VARCHAR(20) | Phone number |
+| `hometown` | VARCHAR(64) | Hometown |
+| `newsletter_opt_in` | BOOLEAN | Newsletter subscription |
+| `created_at` | TIMESTAMP | Account creation time |
+
+#### `votes`
+Individual votes for all contest types.
+
+| Column | Type | Description |
+|--------|------|-------------|
+| `id` | UUID | Primary key |
+| `user_id` | UUID | Foreign key to users (nullable) |
+| `contest_type` | VARCHAR(50) | Contest type enum ('top11', 'year_end_poll', 'mrm') |
```

Comment:
I prefer to make constants all caps - can we use:

- TOP_11
- YEAR_END_POLL
- MODERN_ROCK_MADNESS